sdk/servicebus/service-bus/test/receiveAndDeleteMode.spec.ts

```diff
@@ -349,60 +352,180 @@ describe("receive and delete", () => {
+      );
+    };
+
+    async function testSettlement(
+      testClienttype: TestClientType,
+      operation: DispositionType
+    ): Promise<void> {
+      const deferredMsg = await testDeferredMessage(testClienttype);
+      // we have to force this cast - the type system doesn't allow this if you've chosen receiveAndDelete
+      // as your lock mode.
+      const msg = deferredMsg as ReceivedMessageWithLock;
+
+      try {
+        if (operation === DispositionType.complete) {
+          await msg.complete();
+        } else if (operation === DispositionType.abandon) {
+          await msg.abandon();
+        } else if (operation === DispositionType.deadletter) {
+          await msg.deadLetter();
+        } else if (operation === DispositionType.defer) {
+          await msg.defer();
+        }
+      } catch (err) {
+        errorWasThrown = true;
+        testError(err, operation);
+      }
+
+      should.equal(errorWasThrown, true, "Error thrown flag must be true");
+    }
+
+    it(noSessionTestClientType + ": complete() throws error", async function(): Promise<void> {
+      await testSettlement(noSessionTestClientType, DispositionType.complete);
+    });
+
+    it(withSessionTestClientType + ": complete() throws error", async function(): Promise<void> {
+      await testSettlement(withSessionTestClientType, DispositionType.complete);
+    });
+
+    it(noSessionTestClientType + ": abandon() throws error", async function(): Promise<void> {
+      await testSettlement(noSessionTestClientType, DispositionType.abandon);
+    });
+
+    it(withSessionTestClientType + ": abandon() throws error", async function(): Promise<void> {
+      await testSettlement(withSessionTestClientType, DispositionType.abandon);
+    });
+
+    it(noSessionTestClientType + ": defer() throws error", async function(): Promise<void> {
+      await testSettlement(noSessionTestClientType, DispositionType.defer);
+    });
+
+    it(withSessionTestClientType + ": defer() throws error", async function(): Promise<void> {
+      await testSettlement(withSessionTestClientType, DispositionType.defer);
+    });
+
+    it(noSessionTestClientType + ": deadLetter() throws error", async function(): Promise<void> {
+      await testSettlement(noSessionTestClientType, DispositionType.deadletter);
+    });
+
+    it(withSessionTestClientType + ": deadLetter() throws error", async function(): Promise<void> {
+      await testSettlement(withSessionTestClientType, DispositionType.deadletter);
+    });
+
+    async function testRenewLock(testClienttype: TestClientType): Promise<void> {
+      const deferredMsg = await testDeferredMessage(testClienttype);
+      // we have to force this cast - the type system doesn't allow this if you've chosen receiveAndDelete
+      // as your lock mode.
+
+      // have to cast it - the type system doesn't allow us to call into this method otherwise.
```

Comment:
Repetitive?
```suggestion
      // we have to force this cast - the type system doesn't allow us to call into this method otherwise.
```

Reply:
no idea, I copied the entire test set from the ones we have for "normal" messages in this file

Reply:
Yeah, the problem is that the Receiver isn't being typed more specifically. 

We should fix this at some point to allow the receiver to be scoped properly but I think (grossly) this is the best we can do for now.